webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        let ticketStatusClass = 'default-style';
+
+        const myStatusClass = myStatusClasses[ticketStatus.toLowerCase()];
+        if (myStatusClass) {
+            ticketStatusClass += myStatusClass;
+        } else {
+            ticketStatusClass += ' ticket-status--default';
+        }
+
+        return <span className={ticketStatusClass}>{ticketStatus}</span>;
+    }
+
+    labelList(labels: string[]) {
+        if (!labels.length) {
+            return null;
+        }
+
+        let totalString = 0;
+        let totalHide = 0;
+        const labelList = labels.map((label: string, key: number): JSX.Element => {
+            if (totalString < 3) {
+                totalString++;
+                return (
+                    <span
+                        key={key}
+                        className='popover-labels__label-list'
+                    >
+                        {label}
+                    </span>);
+            }
+
+            totalHide++;
+            if (key === labels.length - 1) {
+                const moreLabels = `+${totalHide}more`;
+                return (
+                    <span
+                        key={key}
+                        className='popover-labels__label-list'
+                    >
+                        {moreLabels}
+                    </span>);
+            }
+
+            return <></>;
+        });
+        return (<div className='popover-labels__label'>{labelList}</div>);
+    }
+
+    render() {
+        if (!this.state.isLoaded) {
+            return (<p/>);
+        }
+
+        const {ticketDetails, href: jiraTicketURI} = this.state;
+
+        let jiraAvatar;
+        let jiraIssueIconURI;
+        let jiraTicketKey;
+        let jiraTicketTitle;
+        let jiraTicketAssigneeAvatarURI;
+        let jiraTicketAssigneeName;
+        let jiraTicketStatusName;
+        let jiraTicketDescription;
+        let jiraTicketVersions;
+        let jiraTicketLabels;
+        if (ticketDetails) {
+            jiraAvatar = ticketDetails.jiraIcon;
+            jiraIssueIconURI = ticketDetails.issueIcon;
+            jiraTicketKey = ticketDetails.ticketId;
+            jiraTicketTitle = ticketDetails.summary;
+            jiraTicketAssigneeAvatarURI = ticketDetails.assigneeAvatar;
+            jiraTicketAssigneeName = ticketDetails.assigneeName;
+            jiraTicketStatusName = ticketDetails.statusKey;
+            jiraTicketDescription = ticketDetails.description;
+            jiraTicketVersions = ticketDetails.versions;
+            jiraTicketLabels = ticketDetails.labels;
+        }
```

Comment:
This seems like an anti-pattern. I'm thinking accessing `ticketDetails` directly in the rendering parts may be better, unless you have reasons why this is better. What do you think @raghavaggarwal2308?